api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -59,17 +81,520 @@ describe('assessmentsRouter', () => {
+        title: 'Assignment 1: React',
+        description: 'Your assignment for week 1 learning.',
+        max_score: 10,
+        max_num_submissions: 3,
+        time_limit: 120,
+        curriculum_id: 3,
+        activity_id: 97,
+        principal_id: facilitatorPrincipalId,
+        questions: [
+          {
+            id: 1,
+            assessment_id: curriculumAssessmentId,
+            title: 'What is React?',
+            description: '',
+            question_type: 'single choice',
+            answers: [
+              {
+                id: 1,
+                question_id: 1,
+                title: 'A relational database management system',
+                description: '',
+                sort_order: 1,
+                correct_answer: true,
+              },
+            ],
+            correct_answer_id: 1,
+            max_score: 1,
+            sort_order: 1,
+          },
+        ],
+      };
+      const programAssessment = {
+        id: programAssessmentId,
+        program_id: 1,
+        assessment_id: curriculumAssessmentId,
+        available_after: '2023-02-06',
+        due_date: '2023-02-10',
+      };
+
+      const assessmentSubmission = {
+        id: 2,
+        assessment_id: programAssessmentId,
+        principal_id: participantPrincipalId,
+        assessment_submission_state: 'graded',
+        score: 1,
+        opened_at: '2023-02-09 12:00:00',
+        submitted_at: '2023-02-09 13:23:45',
+        responses: [
+          {
+            id: 1,
+            answer_id: 1,
+            assessment_id: 1,
+            submission_id: 2,
+            question_id: 1,
+          },
+        ],
+      };
+      const response: SubmittedAssessment = {
+        curriculum_assessment: curriculumAssessment,
+        program_assessment: programAssessment,
+        submission: assessmentSubmission,
+      };
+
+      mockPrincipalId(participantPrincipalId);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+
+        .expect(200, itemEnvelope(response), err => {
+          expect(mockProgramAssessmentById).toHaveBeenCalledWith(
+            programAssessmentId
+          );
+          expect(mockGetCurriculumAssessmentById).toHaveBeenCalledWith(
+            programAssessmentId,
+            true,
+            true
+          );
+
+          expect(mockFindRoleInProgram).toHaveBeenCalledWith(
+            facilitatorPrincipalId,
+            programAssessment.program_id
+          );
+          expect(mockSubmissionDetails).toHaveBeenCalledWith(
+            programAssessmentId,
+            submissionId,
+            true
+          );
+          done(err);
+        });
+    });
+
+    it('should respond with a bad request error if given an invalid assessment id', done => {
+      const programAssessmentId = 'test',
+        submissionId = 1;
+
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(400, done);
```

Comment:
Do you want to do the errorEnvelope thing for all the tests that throw an error, or do you only want to do that for the tests that throw an error after calling service file functions?